src/Core/Form/IdentifiableObject/DataProvider/FormDataProviderInterface.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataProvider;
+
+/**
+ * Interface IdentifiableObjectFormDataProviderInterface.
+ */
+interface FormDataProviderInterface
+{
+    /**
+     * Get form data for given object with given id.
+     *
+     * @param int $id
+     *
+     * @return mixed
```

Comment:
forms accepts any type of data, it can be array, null, object & etc. so i didnt want to limit it to arrays only, wdyt?

Reply:
yes I thought so too, I'm hesitating should we "force" an array in order to guide the developers
if you we start using array in some place, objects in others this could get ugly